pkg/cloudprovider/aws/aws.go

```diff
@@ -73,12 +107,65 @@ func (f *ec2InstanceFilter) Matches(instance ec2.Instance) bool {
 }
 
 // goamzEC2 is an implementation of the EC2 interface, backed by goamz
-type GoamzEC2 struct {
-	ec2 *ec2.EC2
+type goamzEC2 struct {
+	auth aws.Auth
+	ec2  *ec2.EC2
+
+	mutex      sync.Mutex
+	elbClients map[string]*elb.ELB
+}
+
+func newGoamzEC2(auth aws.Auth, regionName string) (*goamzEC2, error) {
+	region, ok := aws.Regions[regionName]
+	if !ok {
+		return nil, fmt.Errorf("not a valid AWS region: %s", regionName)
+	}
+
+	self := &goamzEC2{}
+	self.ec2 = ec2.New(auth, region)
+	self.auth = auth
+	self.elbClients = make(map[string]*elb.ELB)
+	return self, nil
+}
+
+// Find the kubernetes vpc
```

Comment:
Good point.  Fixed the comment.

I do want to support multiple k8s deployments into the same region, which doesn't work right now (#6006).  But that'll be a separate PR.